cosmos/airflow/graph.py

```diff
@@ -409,6 +410,21 @@ def _get_dbt_dag_task_group_identifier(dag: DAG, task_group: TaskGroup | None) -
     return dag_task_group_identifier
 
 
+def calculate_detached_node_name(node: DbtNode, counter: list[int] = [0]) -> str:
+    """
+    Given a detached test node, calculate its name. It will either be:
+     - the name of the test with a "_test" suffix, if this is smaller than 250
+     - or detached_{an incremental number}_test
+    """
```

Comment:
```suggestion
    """
    if counter is None:
        counter = [0]
```

Reply:
ahh I see. Would it be possible to initialise the counter to [0] in the calling function before calling for the first time and then reusing the list somehow while calling it in the subsequent calls? I am not sure if there could be some side-effects due to this. Meaning, the counter would keep on incrementing always across parses of all the DAGs until a restart of the worker, no? 

Also happy to keep as is and observe and perhaps we could refactor later if needed.

Reply:
That's a very good question, @pankajkoti .

The counter is restarted each time a new process is created.

From a scheduler DAG parsing perspective, the `DagFileProcessorManager` process executes an infinite loop that determines which files need to be processed, and it spawns separate `DagFileProcessorProcess` processes that are started to convert an individual file into one or more DAG objects. So we should be safe:
https://airflow.apache.org/docs/apache-airflow/stable/authoring-and-scheduling/dagfile-processing.html

From a worker node perspective, each time a task scheduled, we're running a new process to reparse the DAG and then execute it, so I also believe we're safe.

Reply:
yes, thanks for sharing the link Tatiana. I believe with that we're safe.

Reply:
BTW: just giving additional context, I checked with @ashb, and he confirmed that this applies to Airflow 2.x, and it may apply to Airflow 3.x, but he pointed out - and we all agree - that we're using an internal implementation detail of Airflow that may change.

I've logged a follow up ticket for us to review and improve this further: #1469

Reply:
I also refactored so we don't rely on Python's [default mutable object behaviour](https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil) in 7335ef6bb9f45104ddf5d1e873d3482c8f8e5fb3